packages/cactus-plugin-satp-hermes/src/main/proto/cacti/satp/v02/crash_recovery.proto

```diff
@@ -9,5 +9,82 @@ service CrashRecovery   {
   // util RPCs
 
   // step RPCs
+  rpc RecoverV2Message(RecoverMessage) returns (RecoverUpdateMessage);
+  rpc RecoverV2SuccessMessage(RecoverSuccessMessage) returns (google.protobuf.Empty);
+  rpc RollbackV2Message(RollbackMessage) returns (RollbackAckMessage);
+}
+
+message RecoverMessage  {
+  string sessionId = 1;
+  string messageType = 2;
+  string satpPhase = 3;
+  int32 sequenceNumber = 4;
+  bool isBackup = 5;
+  string newIdentityPublicKey = 6;
+  int64 lastEntryTimestamp = 7;
+  string senderSignature = 8;
+}
+
+message RecoverUpdateMessage {
+  string sessionId = 1;
+  string messageType = 2;
+  string hashRecoverMessage = 3;
+  repeated LocalLog recoveredLogs = 4;
+  string senderSignature = 5;
+}
+
+message RecoverSuccessMessage {
+  string sessionId = 1;
+  string messageType = 2;
+  string hashRecoverUpdateMessage = 3;
+  bool success = 4;
+  repeated string entriesChanged = 5;
+  string senderSignature = 6;
+}
 
+message RollbackMessage {
+  string sessionId = 1;
+  string messageType = 2;
+  bool success = 3;
+  repeated string actionsPerformed = 4;
+  repeated string proofs = 5;
+  string senderSignature = 6;
 }
+
+message RollbackAckMessage {
+  string sessionId = 1;
+  string messageType = 2;
+  bool success = 3;
+  repeated string actionsPerformed = 4;
+  repeated string proofs = 5;
+  string senderSignature = 6;
+}
+
+message LocalLog {
```

Comment:
Please rename to "LogEntry" - this schema should be imported and used by persistLogEntry